File: JavaScript/프로그래머스/4/140105. 쌍둥이 빌딩 숲/쌍둥이 빌딩 숲.js
function solution(n, count) {
    let arr = Array.from({ length: n + 1 }, () => Array(count + 1).fill(0))
    arr[1][1] = 1
    
    for(let i = 2; i <= n ; i++) {
        for(let j = 1; j <= count ; j++) {
            arr[i][j] = arr[i-1][j] * (i-1) * 2 % 1000000007 + arr[i-1][j-1]
            arr[i][j] %= 1000000007
        }
    }
    
    return arr[n][count]
}

File: Python/프로그래머스/4/140105. 쌍둥이 빌딩 숲/쌍둥이 빌딩 숲.py
def solution(n, count):
    dp = [[0] * (count + 1) for _ in range(n + 1)]
    dp[1][1] = 1

    for i in range(2, n + 1):
        for j in range(1, count + 1):
            dp[i][j] = (dp[i - 1][j - 1] + (2 * i - 2) * dp[i - 1][j]) % 1000000007

    return dp[n][count]
